File: PerlinNoise.java
public class PerlinNoise {
    Noise big;
    Noise middle;
    Noise small;
    Noise smaller;
    public PerlinNoise(int width, int height){
        big = new Noise(64, width, height);
        middle = new Noise(32, width, height);
        small = new Noise(16, width, height);
        smaller = new Noise(8, width, height);
    }
    public float getValue(int x, int y){
        return big.geValue(x, y)*0.45f+ middle.geValue(x, y)*0.25f+small.geValue(x, y)*0.15f+smaller.geValue(x,y)*0.15f;
    }
}

File: PerlinNoise.kt
class PerlinNoise(width: Int, height: Int) {
    private val big: Noise = Noise(64, width, height)
    private val middle: Noise = Noise(32, width, height)
    private val small: Noise = Noise(16, width, height)
    private val smaller: Noise = Noise(8, width, height)

    fun getValue(x: Int, y: Int): Float {
        return big.geValue(x, y) * 0.45f + middle.geValue(x, y) * 0.25f + small.geValue(x, y) * 0.15f + smaller.geValue(
            x,
            y
        ) * 0.15f
    }
}
